nxdrive/engine/dao/sqlite.py

```diff
@@ -2660,6 +2660,45 @@ def save_upload(self, upload: Upload) -> None:
             else:
                 self.transferUpdated.emit()
 
+    def save_dt_upload(self, upload: Upload) -> None:
+        """
+        New direct transfer upload.
+        Woll have the same status as it's Session on creation.
+        """
+        with self.lock:
+            # Remove non-serializable data, never used elsewhere
+            batch = {k: v for k, v in upload.batch.items() if k != "blobs"}
+
+            values = (
+                upload.path,
+                upload.doc_pair,
+                TransferStatus.CANCELLED.value,
+                upload.engine,
+                upload.is_direct_edit,
+                upload.is_direct_transfer,
+                upload.filesize,
+                json.dumps(batch),
+                upload.chunk_size,
+                upload.remote_parent_path,
+                upload.remote_parent_ref,
+                upload.doc_pair,
+            )
+            c = self._get_write_connection().cursor()
+            sql = (
+                "INSERT INTO Uploads "
+                "(path, status, engine, is_direct_edit, is_direct_transfer, filesize, batch, chunk_size,"
+                " remote_parent_path, remote_parent_ref, doc_pair)"
+                " VALUES (?, IFNULL((SELECT s.status FROM States INNER JOIN Sessions s ON States.session = s.uid "
+                "AND States.id = ?), ?), ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+            )
+            c.execute(sql, values)
+
+            # Important: update the upload UID and status attr
+            upload.uid = int(c.execute("SELECT last_insert_rowid()").fetchone()[0])
+            res = self.get_dt_upload(uid=upload.uid)
+            upload.status = res.status if res else TransferStatus.CANCELLED
```

Comment:
I do not like that idea. Could we just passe the sessionas argument to prevent requerying the database?